### ISSUE TITLE
Using CORSConfig and setting max_age affects all requests, not just OPTIONS

### ISSUE THREAD

**Issue by leason** [aws/chalice#801]
Reproduction steps:

Create a custom cors config and set the max_age:
`customCors = CORSConfig(
    allow_origin="*",
    max_age=86400
)`

Now use that in a route:
`@app.route('/users/{userId}', methods=['GET'], cors=customCors)`

As expected, the OPTIONS calls get cached because the response header sets `Access-Control-Max-Age` properly.  However, the GET call to /users/{userId} also includes Access-Control-Max-Age in the response headers, which is not expected behavior.  Shouldn't that only be set on the OPTIONS response?

**Comment by stealthycoin** [comment]
Yes all the headers are sent for the preflight request and the real request. This is required for some of the preflight requests but you correctly comment not the `Access-Control-Max-Age`. I can see if this can be finessed a little bit more in configuring API Gateway.

This shouldn't hurt anything though so I'm just going to mark it as an enhancement.
